django/db/models/query.py

```diff
@@ -495,8 +494,7 @@ def _create_object_from_params(self, lookup, params):
         Used by get_or_create and update_or_create
         """
         try:
-            with transaction.atomic(using=self.db):
-                obj = self.create(**params)
+            obj = self.create(**params)
             return obj, True
         except IntegrityError:
```

Comment:
This makes makes it not obvious that all that the contents of the `try/except IntegrityError` are properly wrapped in a transaction.

This is necessary to prevent errors on databases which actually enforce transactional integrity on errors like PostgreSQL.